storage/innobase/handler/ha_innodb.cc

```diff
@@ -2386,7 +2386,7 @@ static int mysql_tmpfile_path(const char *path, const char *prefix)
 
   char filename[FN_REFLEN];
   File fd = create_temp_file(filename, path, prefix, O_BINARY | O_SEQUENTIAL,
-                             MYF(MY_WME | MY_TEMPORARY));
+                             MYF(MY_WME | MY_TEMPORARY | MY_NO_REGISTER));
```

Comment:
I don't really like this. It is InnoDB fault that it handles temp files inconsistently: creates with mysys and then closes directly with libc. InnoDB should use either approach, but not mix them.

Reply:
inconsistency in approaches is fixed by added 5e82fbe